kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java

```diff
@@ -433,4 +433,86 @@ public KaldbSearch.SearchResult doSearch(final KaldbSearch.SearchRequest request
       throw new RuntimeException(e);
     }
   }
+
+  @Override
+  public KaldbSearch.SchemaResult getSchema(KaldbSearch.SchemaRequest distribSchemaReq) {
+    // todo - this shares a significant amount of code with the distributed search request
+    //  and would benefit from refactoring the current "distributedSearch" abstraction to support
+    //  different types of requests
+
+    LOG.info("Starting distributed search for schema request: {}", distribSchemaReq);
+    ScopedSpan span =
+        Tracing.currentTracer().startScopedSpan("KaldbDistributedQueryService.distributedSchema");
+
+    Map<String, SnapshotMetadata> snapshotsMatchingQuery =
```

Comment:
Does this logic need to be same as the query path? Can't it be simpler like - "Query all live nodes" and don't try to find snapshots/search metadata etc?

Reply:
This approach allows us to query historical data and get an appropriate schema set for that time period. We want to take advantage of the same logic where a single snapshot may be on multiple nodes, and be able to include schema that was available an hour ago but may not be appearing now.